src/Native/System.Native/pal_networking.cpp

```diff
@@ -13,6 +13,19 @@
 #include <sys/socket.h>
 #include <unistd.h>
 #include <vector>
+#include <errno.h>
+
+#if defined(__APPLE__)
```

Comment:
Out of curiosity, is this check required? The guard `define`s can be used by other kinds of Unices as well which lack `IPV6_[ADD/DROP]_MEMBERSHIP`.